src/Snippets/Docs/HttpClientIntegrations.cs

```diff
@@ -0,0 +1,129 @@
+using System.Net;
+using System.Net.Http;
+using Flurl.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Polly.Retry;
+using Refit;
+using RestSharp;
+
+namespace Snippets.Docs;
+
+internal static class HttpClientIntegrations
+{
+    private static readonly Uri DownstreamUri = new("https://httpstat.us/408");
+
+    #region http-client-integrations-handle-transient-errors
+    private static ValueTask<bool> HandleTransientHttpError(Outcome<HttpResponseMessage> outcome)
+    => outcome switch
+    {
+        { Exception: HttpRequestException } => PredicateResult.True(),
+        { Result.StatusCode: HttpStatusCode.RequestTimeout } => PredicateResult.True(),
+        { Result.StatusCode: >= HttpStatusCode.InternalServerError } => PredicateResult.True(),
+        _ => PredicateResult.False()
+    };
+
+    private static RetryStrategyOptions<HttpResponseMessage> GetRetryOptions()
+    => new()
+    {
+        ShouldHandle = args => HandleTransientHttpError(args.Outcome),
+        MaxRetryAttempts = 3,
+        BackoffType = DelayBackoffType.Exponential,
+        Delay = TimeSpan.FromSeconds(2)
+    };
+    #endregion
+
+    public static async Task HttpClientExample()
+    {
+        #region http-client-integrations-httpclient
+        ServiceCollection services = new();
+
+        // Register a named HttpClient and decorate with a resilience pipeline
+        services.AddHttpClient(string.Empty)
+                .ConfigureHttpClient(client => client.BaseAddress = DownstreamUri)
+                .AddResilienceHandler("httpclient_based_pipeline",
+                    builder => builder.AddRetry(GetRetryOptions()));
+
+        var provider = services.BuildServiceProvider();
+
+        // Resolve the named HttpClient
```

Comment:
There's no actual name being used here, it's using the default `""` name (which `AddHttpClient()` does). Instead lets pass a name to `AddHttpClient(string)` and then use the same name here.